### ISSUE TITLE
Potential Memory/model slowdown for currentInventory() depending on usage

### ISSUE THREAD

**Issue by BrandonSurowiec** [Caryley/Laravel-Inventory#11]
Glancing through the code, there are a lot of references to `inventories->first()`, which means that as the inventory database rows get added for a product, in some cases Laravel is going to load all database rows / Eloquent models of the product's inventory into memory only to utilize the first model! So the application, in regards to inventory calls, will gradually get slower as the inventory is managed.

The current strategy of `inventories()->first()` works, but if you want to eager load the `currentInventory()` for a bunch of products at once, a bunch of extra models will be loaded into memory for the `inventories` relationship.

This can be improved by leveraging "One of Many" which is in Laravel 8.
https://laravel.com/docs/8.x/eloquent-relationships#one-of-many-polymorphic-relations
https://laravel-news.com/one-of-many-eloquent-relationship

```php
/**
 * Get the user's most recent image.
 */
public function latestImage()
{
    return $this->morphOne(Image::class, 'imageable')->latestOfMany();
}
```

I may end up using this package for a project so if I do, I can help with a PR if you don't get to it first. Hope this information is useful!

**Comment by talelmishali** [comment]
Hi @BrandonSurowiec ,
First of all, thank you for sharing your opinion regarding the package.

I can certainly see where it can be of an issue, I will need to test it a bit further just to make sure it won't break anything (shouldn't). Feel free to PR though.

Tal

**Comment by talelmishali** [comment]
@BrandonSurowiec you can check [develop](https://github.com/Caryley/Laravel-Inventory/tree/develop)

Please feel free to share your thoughts.

**Comment by talelmishali** [comment]
I believe I addressed your concern here #16.
